File: features/xai-oauth/src/xai_oauth/headers.py
from __future__ import annotations

from xai_oauth.constants import TOKEN_AUTH_VALUE, VERSION


def proxy_header_map(access_token: str, version: str = VERSION) -> dict[str, str]:
    identity = f"rhc/{version}"
    return {
        "Authorization": f"Bearer {access_token}",
        "X-XAI-Token-Auth": TOKEN_AUTH_VALUE,
        "User-Agent": identity,
        "x-grok-client-version": identity,
    }

File: crates/rhc/src/host/extensions/auth/headers.rs
use std::collections::BTreeMap;

use super::constants::{TOKEN_AUTH_VALUE, client_identity};

pub fn proxy_header_map(access_token: &str) -> BTreeMap<String, String> {
    let identity = client_identity();
    BTreeMap::from([
        (
            "Authorization".to_string(),
            format!("Bearer {access_token}"),
        ),
        ("X-XAI-Token-Auth".to_string(), TOKEN_AUTH_VALUE.to_string()),
        ("User-Agent".to_string(), identity.clone()),
        ("x-grok-client-version".to_string(), identity),
    ])
}

#[cfg(test)]
#[path = "headers_tests.rs"]
mod tests;
